app/src/components/layout/NavMenu.tsx

```diff
@@ -49,42 +51,44 @@ const Styled = {
 const NavItem: React.FC<{
   page: string;
   badge?: string;
-  onClick: () => void;
-}> = observer(({ page, badge, onClick }) => {
+}> = observer(({ page, badge }) => {
   const { l } = usePrefixedTranslation('cmps.layout.NavMenu');
-  const { router } = useStore();
-  const className = router.location.pathname.startsWith(`${PUBLIC_URL}/${page}`)
-    ? 'active'
-    : '';
+  const { settingsStore, log } = useStore();
+  const location = useLocation();
+  const className = location.pathname.startsWith(`/${page}`) ? 'active' : '';
 
+  const onClick = () => {
+    settingsStore.autoCollapseSidebar();
+    log.info(`Go to the ${page} page`);
+  };
   return (
     <Styled.NavItem className={className}>
-      <span onClick={onClick}>
-        {l(page)}
-        {badge && (
-          <sup>
-            <Badge muted>{badge}</Badge>
-          </sup>
-        )}
-      </span>
+      <Link to={`/${page}`} onClick={onClick}>
```

Comment:
Using `react-router` `Link` for routing as opposed to having a callback that invokes `useNavigate`.

Reply:
I like this approach much better. The `appView.goToXXX` began to smell fishy after it was initially implemented. Now we can open these links in a new tab, which is minor but can be very handy sometimes.

The only drawback is that there is additional logic in `goToLoop` and `goToPool` that is no longer being executed since these functions are not being called. I would suggest to move the logic into `useEffect` hooks on the `LoopPage` and `PoolPage` components. It actually makes more sense there to ensure the code is always executed when the pages are mounted.